app/build.gradle

```diff
@@ -171,82 +174,100 @@ sonarqube {
 }
 
 dependencies {
+/** Desugaring **/
     coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.1'
 
+/** NewPipe libraries **/
+    // You can use a local version by uncommenting a few lines in settings.gradle
+    implementation 'com.github.TeamNewPipe:nanojson:1d9e1aea9049fc9f85e68b43ba39fe7be1c1f751'
+    implementation 'com.github.TeamNewPipe:NewPipeExtractor:def745b801b2ef35c1a0fee1be950331ca6a0cd2'
+
+/** Checkstyle **/
+    checkstyle "com.puppycrawl.tools:checkstyle:${checkstyleVersion}"
+    ktlint 'com.pinterest:ktlint:0.40.0'
+
+/** Kotlin **/
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:${kotlin_version}"
 
+/** AndroidX **/
+    implementation 'androidx.appcompat:appcompat:1.2.0'
+    implementation 'androidx.cardview:cardview:1.0.0'
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
+    implementation 'androidx.core:core-ktx:1.3.2'
+    implementation 'androidx.documentfile:documentfile:1.0.1'
+    implementation "androidx.lifecycle:lifecycle-livedata:${androidxLifecycleVersion}"
+    implementation "androidx.lifecycle:lifecycle-viewmodel:${androidxLifecycleVersion}"
+    implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.0.0'
+    implementation 'androidx.media:media:1.2.1'
+    implementation 'androidx.multidex:multidex:2.0.1'
+    implementation 'androidx.preference:preference:1.1.1'
+    implementation 'androidx.recyclerview:recyclerview:1.1.0'
+    implementation "androidx.room:room-runtime:${androidxRoomVersion}"
+    implementation "androidx.room:room-rxjava3:${androidxRoomVersion}"
+    kapt "androidx.room:room-compiler:${androidxRoomVersion}"
+    implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0'
+    implementation 'androidx.webkit:webkit:1.4.0'
+    implementation 'com.google.android.material:material:1.2.1'
+
+/** Third-party libraries **/
+    // Instance state boilerplate elimination
     implementation "frankiesardo:icepick:${icepickVersion}"
     kapt "frankiesardo:icepick-processor:${icepickVersion}"
 
-    checkstyle "com.puppycrawl.tools:checkstyle:${checkstyleVersion}"
-    ktlint "com.pinterest:ktlint:0.40.0"
-
-    debugImplementation "com.facebook.stetho:stetho:${stethoVersion}"
-    debugImplementation "com.facebook.stetho:stetho-okhttp3:${stethoVersion}"
-
-    debugImplementation "com.squareup.leakcanary:leakcanary-android:${leakCanaryVersion}"
+    // Memory leak detection
```

Comment:
This is also related to debugging and should go below

Reply:
I actually have those separate because they're only built for a specific build type, but I guess I can move them into the debugging section